### ISSUE TITLE
"Error: KafkaConsumer is not connected" while disconnecting

### ISSUE THREAD

**Issue by oliviertassinari** [Blizzard/node-rdkafka#227]
I have been using the steam API.
I have noticed that an error is, from time to time, thrown during the disconnection.
I use the following work around for now:
```js
stream.on('error', (errorMessage) => {
  if (stream.consumer._isDisconnecting) {
    return
  }

  // Normal error logic handling
  // ...
})

stream.close()
```

**Comment by michallevin** [comment]
@oliviertassinari I am trying your workaround but the error is still being thrown... 🤔

**Comment by webmakersteve** [comment]
I think this is fixed with #321
